### PR TITLE
191 flexible validation error console height

### PR DESCRIPTION
**Pull request by roshan-gh**

### CODE REVIEW

File: packages/otelbin/src/components/monaco-editor/Editor.tsx [hunk 1]
```diff
@@ -179,7 +179,7 @@ export default function Editor({ locked, setLocked }: { locked: boolean; setLock
 							)}
 						</div>
 						{viewMode !== "pipeline" && <ValidationErrorConsole errors={totalValidationErrors} font={firaCode} />}
-						{viewMode == "both" && <ResizeBar onWidthChange={onWidthChange} />}
```
Comment (bripkens) [inline packages/otelbin/src/components/monaco-editor/Editor.tsx:182]: **Question:** Why was this condition removed? Now the resize bar is shown when the editor is hidden.

File: packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx [hunk 2]
```diff
@@ -42,58 +51,123 @@ export default function ValidationErrorConsole({ errors, font }: { errors?: IErr
 		}
 	}, [errorCount, warningsCount]);
 
+	const onHeightChange = useCallback((newHeight: number) => {
```
Comment (bripkens) [inline packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx:54]: **Question:** Why do we need this `useCallback` fn? As far as I can see, we could just use `setHeight` within the `useEffect` call below.

File: packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx [hunk 3]
```diff
@@ -42,58 +51,123 @@ export default function ValidationErrorConsole({ errors, font }: { errors?: IErr
...
-							})}
-						{errors?.customErrors &&
-							errors.customErrors?.length > 0 &&
-							errors.customErrors.map((error: string, index: number) => {
-								return <ErrorMessage key={index} customErrors={error} font={font} />;
-							})}
-						{errors?.jsYamlError?.mark?.line && <ErrorMessage jsYamlError={errors?.jsYamlError} font={font} />}
-					</div>
-				)}
+		<>
```
Comment (bripkens) [inline packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx:94]: **Question:** Why do we need this fragment?

File: packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx (hunk 2 above)
Comment (roshan-gh) [reply packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx]: fixed and pushed. @bripkens

File: packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx (hunk 3 above)
Comment (roshan-gh) [reply packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx]: fixed and pushed. @bripkens

File: packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx [hunk 4]
```diff
@@ -42,57 +51,116 @@ export default function ValidationErrorConsole({ errors, font }: { errors?: IErr
...
-								return <ErrorMessage key={index} ajvError={error} font={font} />;
-							})}
-						{errors?.customErrors &&
-							errors.customErrors?.length > 0 &&
-							errors.customErrors.map((error: string, index: number) => {
-								return <ErrorMessage key={index} customErrors={error} font={font} />;
-							})}
-						{errors?.jsYamlError?.mark?.line && <ErrorMessage jsYamlError={errors?.jsYamlError} font={font} />}
+			<AutoSizer>
+				{({ height }) => (
```
Comment (bripkens) [inline packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx:108]: **Question:** Why do we need `AutoSizer` here?

Comment (roshan-gh) [reply packages/otelbin/src/components/monaco-editor/ValidationErrorConsole.tsx]: @bripkens at initial implementation I used it to fix transition and lag problem. Now removed it and pushed. thanks for notice. @bripkens

**Pull request closed without merging**